src/seraphis_wallet/serialization_types.cpp

```diff
@@ -0,0 +1,137 @@
+//    materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "serialization_types.h"
+
+// local headers
+#include "ringct/rctTypes.h"
+#include "seraphis_core/jamtis_destination.h"
+#include "seraphis_crypto/sp_crypto_utils.h"
+#include "seraphis_impl/enote_store.h"
+#include "seraphis_impl/serialization_demo_types.h"
+#include "seraphis_impl/serialization_demo_utils.h"
+#include "seraphis_main/contextual_enote_record_types.h"
+#include "seraphis_main/sp_knowledge_proof_types.h"
+#include "seraphis_main/sp_knowledge_proof_utils.h"
+
+// third party headers
+#include <boost/range.hpp>
+
+#include "boost/range/iterator_range.hpp"
+#include "seraphis_wallet/transaction_history.h"
+#include "serialization/binary_archive.h"
+#include "serialization/containers.h"
+#include "serialization/serialization.h"
+
+// standard headers
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <map>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+void make_serializable_transaction_record_v1(const TransactionRecordV1 &tx_rec, ser_TransactionRecordV1 &ser_tx_rec)
+{
+    ser_tx_rec.legacy_spent_enotes = tx_rec.legacy_spent_enotes;
+    ser_tx_rec.sp_spent_enotes = tx_rec.sp_spent_enotes;
+    sp::serialization::ser_JamtisDestinationV1 ser_dest{};
+    std::vector<std::pair<sp::serialization::ser_JamtisDestinationV1, rct::xmr_amount>> outlays_aux{};
+    for (auto &r : tx_rec.outlays)
+    {
+        make_serializable_sp_destination_v1(r.first, ser_dest);
+        outlays_aux.emplace_back(ser_dest, r.second);
+    }
+    ser_tx_rec.outlays = outlays_aux;
+    ser_tx_rec.amount_sent = tx_rec.amount_sent;
+    ser_tx_rec.fee_sent = tx_rec.fee_sent;
+    outlays_aux.clear();
+}
+
+void make_serializable_sp_transaction_store_v1(const SpTransactionStoreV1 &tx_store,
+                                               ser_SpTransactionStoreV1 &ser_tx_store)
+{
```

Comment:
`ser_SpTransactionStoreV1` is a quite strange name.

It starts with a lowercase letter which looks quite strange to me.

It mixes two naming styles, using underscores to separate words, and camel case.

In the Seraphis library only few words within names are abbreviated, but here we have `ser` instead of `Serialized`.

I checked the "Serialization demo" files in the Seraphis library, and much to my surprise, that `ser_` prefix was used by @UkoeHB there. Still, with the naming in the library usually so systematic and well thought-out, I wonder whether this `ser_` is merely there because so far that code is only that, a **demo**.